web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -482,7 +482,8 @@ export function UnifiedResources(props: UnifiedResourcesProps) {
             )}
             isProcessing={
               resourcesFetchAttempt.status === 'processing' ||
-              getPinnedResourcesAttempt.status === 'processing'
+              getPinnedResourcesAttempt.status === 'processing' ||
+              unifiedResourcePreferencesAttempt.status === 'processing'
```

Comment:
I'm not 100% sure, but those checks should probably check `''` as well to show the loading skeleton ASAP. Otherwise they'll wait for `useEffect` or something else to start fetching and change the state from `''` to `'processing'`. Unless the hook for fetching the resources has some extra logic about turning the attempt status back to `''` when it's done, idk.

Reply:
Yeah, `useKeyBasedPagination` has that logic unfortunately, but I added it for other cases.